controller/linodemachine_controller_helpers.go

```diff
@@ -35,16 +37,62 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
+
+	infrav1alpha1 "github.com/linode/cluster-api-provider-linode/api/v1alpha1"
 )
 
+func (*LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineScope *scope.MachineScope, tags []string, logger logr.Logger) (*linodego.InstanceCreateOptions, error) {
+	var err error
+
+	createConfig := linodeMachineSpecToInstanceCreateConfig(machineScope.LinodeMachine.Spec)
+	if createConfig == nil {
+		err = errors.New("failed to convert machine spec to create instance config")
+
+		logger.Error(err, "Panic! Struct of LinodeMachineSpec is different than InstanceCreateOptions")
+
+		return nil, err
+	}
+	createConfig.SwapSize = util.Pointer(0)
+	createConfig.PrivateIP = true
+
+	bootstrapData, err := machineScope.GetBootstrapData(ctx)
+	if err != nil {
+		logger.Info("Failed to get bootstrap data", "error", err.Error())
+
+		return nil, err
+	}
+	createConfig.Metadata = &linodego.InstanceMetadataOptions{
+		UserData: b64.StdEncoding.EncodeToString(bootstrapData),
```

Comment:
There's a size limit of 65535 bytes, should we check to ensure that's not exceeded?

Reply:
can't we gzip it too?

Reply:
@luthermonson @avestuk @AshleyDumaine I created a ticket for this change, because i feel it is out of the scope of this PR.

Reply:
Works for me